crates/yttrium/src/sign/mod.rs

```diff
@@ -415,19 +421,19 @@ impl Client {
                 method: "wc_sessionSettle".to_string(),
                 params: SessionSettle {
                     relay: Relay { protocol: "irn".to_string() },
-                    namespaces,
+                    namespaces: approved_namespaces,
                     controller: Controller {
                         public_key: hex::encode(self_public_key.to_bytes()),
-                        metadata,
+                        metadata: self_metadata,
                     },
-                    expiry_timestamp: crate::time::SystemTime::now()
+                    expiry: crate::time::SystemTime::now()
                         .duration_since(crate::time::UNIX_EPOCH)
                         .unwrap()
                         .as_secs()
-                        + 60 * 60 * 24 * 30, // TODO
-                    session_properties: serde_json::Value::Null, // TODO
-                    scoped_properties: serde_json::Value::Null,  // TODO
-                    session_config: serde_json::Value::Null,     // TODO
+                        + 60 * 60 * 24 * 7, //Session expiry is 7 days
+                    session_properties: proposal.session_properties.as_ref().map(|p| serde_json::to_value(p).unwrap_or_default()).unwrap_or_default(),
+                    scoped_properties: proposal.scoped_properties.as_ref().map(|p| serde_json::to_value(p).unwrap_or_default()).unwrap_or_default(),
+                    // session_config: proposal.session_config,
```

Comment:
Are you sure you want `or_default()` here? This means that in the error case, the default value would be put here. E.g. `Null` or empty maps.

I suspect it would be better to explicitly error if there is an issue here, vs defensively trying to still make it work

Reply:
Actually here it makes sense to use or_default because these values might not be set, they're optional so setting defaults make sense to me